site/_posts/2020-11-15-how-to-securely-implement-oauth-rails.md

```diff
@@ -0,0 +1,300 @@
+* **Email**
+* **Password and Confirm**: Toggle `Send email to setup password` to off and manually enter a password for the user.
+
+### Register the User
+Looking good! Now we have our new application and a user. The last thing we want to do is `register` the user with our `application`.
+Again, navigate to the `Users` page, find our newly created user, and click the `Manage` button.
+
+Click on the `Add Registration` button, select the Rails application we created, and save.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/user-registered.png" alt="The Manage User page in FusionAuth." class="img-fluid" figure=false %}
+
+## Build the Rails Application
+Lets get started building a very basic Rails application to demonstrate authentication with FusionAuth.
+
+Create a new Rails application.
+```bash
+rails new rails-fusionauth-app
+```
+
+Our `Gemfile` will use a few additional dependencies so go ahead and add them before we get started.
+```ruby
+gem 'jwt'
+gem 'oauth'
+```
+
+Install dependencies
+```bash
+bundle install
+```
+
+### Configuration
+Before we go any further, there are some critical `Application` attributes we will need from our FusionAuth configuration so that our Rails app knows how to successfully communicate with it.
+Specifically:
+* `client_id`
+* `client_secret`
+* `Authorized redirect URL's`
+* `Logout URL`
+
+You can recall, these can be found under the `OAuth` tab when modifying an application in FusionAuth.
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/edit-application.png" alt="The OAuth tab in the Edit Application page of FusionAuth." class="img-fluid" figure=false %}
+
+### Routes
+For this app, we are going to setup three routes:
+* **welcome page**: Our root page or `view`. 
+* **oauth2-callback** endpoint: Receiver of the redirect after logging into FusionAuth.
+* **logout** endpoint: Receiver of the redirect after logging out of FusionAuth.
+
+```ruby
+Rails.application.routes.draw do
+  root to: 'welcome#index'
+  get '/oauth2-callback', to: 'oauth#oauth_callback'
+  get '/logout', to: 'oauth#destroy'
+end
+```
+
+### Authenticating and Logging Out 
+We will start off with a very basic welcome view that contains a link for a user to login to the application.
+
+```ruby
+# app/views/welcome/index.html.erb
+
+<p>Welcome user! Please <%= link_to 'Log In', @login_url %></p>
+```
+
+It should look something like this when rendered.
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-rails/home.png" alt="The example Home screen for the Rails application." class="img-fluid" figure=false %}
+
+To begin the login process, we will want to construct a URL to make an [Authorization Code Grant Request](https://fusionauth.io/docs/v1/tech/oauth/endpoints/#authorization-code-grant-request).
+This URL is composed of the `client_id`, `client_secret`, and `redirect_uri` values from our application configuration. 
```

Comment:
What is the difference between redirect_uri and authorized redirect URLs? Maybe explain that? Might be confusing.